inbm/dispatcher-agent/dispatcher/dispatcher_class.py

```diff
@@ -210,7 +210,7 @@ def _perform_startup_tasks(self) -> None:
                         create_snapshotter('update',
                                            snap_num='1',
                                            proceed_without_rollback=True,
-                                           ).commit()
+                                           device_reboot='Y').commit()
```

Comment:
**Suggestion:** The 'device_reboot' parameter should respect the 'proceed_without_rollback' flag and the 'reboot' setting from the manifest. Hardcoding it to 'Y' may cause unintended reboots when the manifest specifies 'reboot=no'. Consider using a variable to control this behavior based on the manifest settings. [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                        create_snapshotter('update',
                                           snap_num='1',
                                           proceed_without_rollback=True,
                                           device_reboot=self._should_reboot()).commit()
```